src/steps/os/openbsd.rs

```diff
@@ -1,16 +1,31 @@
 use crate::command::CommandExt;
 use crate::execution_context::ExecutionContext;
+use crate::t;
 use crate::terminal::print_separator;
 use crate::utils::{get_require_sudo_string, require_option};
 use color_eyre::eyre::Result;
+use std::process::Command;
+
+fn is_openbsd_current() -> bool {
+    let output = Command::new("uname")
+        .arg("-r")
+        .output()
+        .expect("Failed to execute uname command");
+
+    let version = String::from_utf8_lossy(&output.stdout);
+    version.trim().ends_with("-current")
+}
 
 pub fn upgrade_openbsd(ctx: &ExecutionContext) -> Result<()> {
     let sudo = require_option(ctx.sudo().as_ref(), get_require_sudo_string())?;
     print_separator(t!("OpenBSD Update"));
-    ctx.run_type()
-        .execute(sudo)
-        .args(["/usr/sbin/sysupgrade", "-n"])
-        .status_checked()
+
+    let mut args = vec!["/usr/sbin/sysupgrade", "-n"];
+    if is_openbsd_current() {
```

Comment:
This function uses std::process::Command and Topgrade's dry-run option does not work with it, and thus this PR breaks the dry-run option.

We should use ctx.run_type().execute() instead, though I kinda think it won't work either. 😮‍💨

I guess we should skip all the update procedures and print something like "will update the system package" if this is a dry run(ctx has a dry_run() function to check if the current run is a dry-run or not)